rviz_default_plugins/src/rviz_default_plugins/displays/camera/camera_display.cpp

```diff
@@ -545,5 +553,13 @@ void CameraDisplay::reset()
 
 }  // namespace rviz_default_plugins
 
+#ifndef _WIN32
+# pragma GCC diagnostic push
+# pragma GCC diagnostic ignored "-Wunused-parameter"
+# pragma GCC diagnostic ignored "-Wpedantic"
+#endif
```

Comment:
Why is this needed?

Reply:
We should only disable warnings like this when they're in external dependencies. This looks like an issue within pluginlib?

Reply:
I totally agree, I just wanted to see whether I could get it to compile if I did this.

I get the following warning: 
```
/include/pluginlib/class_list_macros.hpp:49:59: error: extra ‘;’ [-Werror=pedantic]
   CLASS_LOADER_REGISTER_CLASS(class_type, base_class_type);
```
If I read this correctly, it complains that the ";" is superfluous in this line:

https://github.com/ros/pluginlib/blob/b456beba88564b746694dde68696d8bc4edc78c9/pluginlib/include/pluginlib/class_list_macros.hpp#L49

It seems to me that compiling the pluginlib with -Wpedantic should catch that problem.